### PR TITLE
Add dynamic options to pd_form

### PR DESCRIPTION
**Pull request by aoby**

### CODE REVIEW

File: dashboard/app/models/concerns/pd/form.rb [hunk 1]
```diff
@@ -39,7 +44,7 @@ def validate_required_fields
     # empty fields may come about when the user selects then unselects an
     # option. They should be treated as if they do not exist
     hash.delete_if do |_, value|
-      value.empty?
+      value.blank?
```
Comment (Hamms) [inline dashboard/app/models/concerns/pd/form.rb:51]: why this change?

Comment (aoby) [reply dashboard/app/models/concerns/pd/form.rb:51]: I encountered an error at some point during development because I had a nil (rather than empty) value. I don't remember how I got there, but `blank?` covers `nil` and `empty?`.

Is there a reason we'd want to preserve nil values?

**Comment by aoby** [comment]
Good suggestion. I'll add a comment

File: dashboard/app/models/concerns/pd/form.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/app/models/concerns/pd/form.rb:51]: ooooh right, `nil`. Good call
